test/e2e/empty_dir_wrapper.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/intstr"
+
+	"strconv"
+
+	. "github.com/onsi/ginkgo"
+)
+
+// This test will create a pod with a secret volume and gitRepo volume
+// Thus requests a secret, a git server pod, and a git server service
+var _ = Describe("EmptyDir wrapper volumes", func() {
+	f := NewFramework("secrets")
+
+	It("should becomes running [Conformance]", func() {
+		name := "secret-test-" + string(util.NewUUID())
+		volumeName := "secret-volume"
+		volumeMountPath := "/etc/secret-volume"
+
+		secret := &api.Secret{
+			ObjectMeta: api.ObjectMeta{
+				Namespace: f.Namespace.Name,
+				Name:      name,
+			},
+			Data: map[string][]byte{
+				"data-1": []byte("value-1\n"),
+			},
+		}
+
+		var err error
+		if secret, err = f.Client.Secrets(f.Namespace.Name).Create(secret); err != nil {
+			Failf("unable to create test secret %s: %v", secret.Name, err)
+		}
+
+		gitServerPodName := "git-server-" + string(util.NewUUID())
+
+		labels := map[string]string{"name": gitServerPodName}
+
+		gitServerPod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   gitServerPodName,
+				Labels: labels,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:            "git-repo",
+						Image:           "hyzual/gitolite-http",
```

Comment:
We try not to rely on 3rd party images.  We can pull it, tag it, and push
to gcr
On Jan 12, 2016 8:05 AM, "Paul Morie" <notifications@github.com> wrote:

> In test/e2e/empty_dir_wrapper.go
> <https://github.com/kubernetes/kubernetes/pull/18445#discussion_r49473359>
> :
>
> > +		}
> > +
> > +		gitServerPodName := "git-server-" + string(util.NewUUID())
> > +
> > +		labels := map[string]string{"name": gitServerPodName}
> > +
> > +		gitServerPod := &api.Pod{
> > +			ObjectMeta: api.ObjectMeta{
> > +				Name:   gitServerPodName,
> > +				Labels: labels,
> > +			},
> > +			Spec: api.PodSpec{
> > +				Containers: []api.Container{
> > +					{
> > +						Name:            "git-repo",
> > +						Image:           "hyzual/gitolite-http",
>
> @thockin <https://github.com/thockin> Do we have any kind of policy on
> using third-party images in E2Es?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18445/files#r49473359>.
>